src/coreclr/jit/emitarm64.cpp

```diff
@@ -10072,7 +10111,18 @@ void emitter::emitIns_R_R_I(instruction     ins,
 
     assert(fmt != IF_NONE);
 
-    instrDesc* id = emitNewInstrSC(attr, imm);
+    instrDesc* id;
+
+    if (!optionalShift)
```

Comment:
I am little worried about the TP impact because of this check.

Reply:
hhm, but looks like we did similar thing for `emitIns_R_I()` as well.

Reply:
I remember adding this in `emitIns_R_I`. We don't have the TP results from that PR anymore, but I remember it being nontrivial. This might be another reason to separate out the SVE encodings into `emitIns_SVE` methods later.

Reply:
> This might be another reason to separate out the SVE encodings into `emitIns_SVE` methods later.

If the TP results are changed, then I'll do this.

Reply:
I don't see anything posted from the bot yet, I assume there is just a backlog?

Reply:
> I don't see anything posted from the bot yet, I assume there is just a backlog?

`dotnet-bot` is not running temporarily, but you can see the diffs on CI pipeline (the way you would see before bot came).

Reply:
I'll check this once the tests have all re-run

Reply:
No obvious TP issues, but we should probably split out in a later PR.

Reply:
I'm planning on cleaning up our usage of `idOptionalShift` after this PR is merged, as that will simplify my last few encodings. I think we can save the immediate in its decoded state, and encode the shift state later in `emitOutputInstr`. It will make displaying the instruction slightly less elegant, but by getting rid of the `idOptionalShift` state, we can get rid of this branch and always call `emitNewInstrSC`.

Reply:
That would be great. There's possibly a few more immediates than can get squashed too.